integration_tests/src/main/python/url_test.py

```diff
@@ -0,0 +1,234 @@
+    # "http://[::]",
+    # "http://[fe80::7:8%eth0]",
+    # "http://[fe80::7:8%1]",
+    # "http://[::255.255.255.255]",
+    # "http://[::ffff:255.255.255.255]",
+    # "http://[::ffff:0:255.255.255.255]",
+    # "http://[2001:db8:3:4::192.0.2.33]",
+    # "http://[64:ff9b::192.0.2.33]"
+]
+
+edge_cases_gen = SetValuesGen(StringType(), edge_cases)
+
+url_gen = StringGen(url_pattern)
+    
+def test_parse_url_protocol():
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, url_gen).selectExpr(
+                "a",
+                "parse_url(a, 'PROTOCOL')"
+                ))
+    
+def test_parse_url_protocol_edge_cases():
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, edge_cases_gen).selectExpr(
+                "a",
+                "parse_url(a, 'PROTOCOL')"
+                ))
+
+@allow_non_gpu('ProjectExec', 'ParseUrl')
+def test_parse_url_host_fallback():
+    assert_gpu_fallback_collect(
+            lambda spark : unary_op_df(spark, url_gen).selectExpr(
+                "a",
+                "parse_url(a, 'HOST')"
+                ),
+            'ParseUrl')
+    
+@allow_non_gpu('ProjectExec', 'ParseUrl')
+def test_parse_url_path_fallback():
+    assert_gpu_fallback_collect(
+            lambda spark : unary_op_df(spark, url_gen).selectExpr(
+                "a",
+                "parse_url(a, 'PATH')"
+                ),
+            'ParseUrl')
+
+@allow_non_gpu('ProjectExec', 'ParseUrl')
+def test_parse_url_query_fallback():
+    assert_gpu_fallback_collect(
+            lambda spark : unary_op_df(spark, url_gen).selectExpr(
+                "a",
+                "parse_url(a, 'QUERY')"
+                ),
+            'ParseUrl')
+
+@allow_non_gpu('ProjectExec', 'ParseUrl')   
+def test_parse_url_ref_fallback():
+    assert_gpu_fallback_collect(
+            lambda spark : unary_op_df(spark, url_gen).selectExpr(
+                "a",
+                "parse_url(a, 'REF')"
+                ),
+            'ParseUrl')
+
+@allow_non_gpu('ProjectExec', 'ParseUrl')  
+def test_parse_url_file_fallback():
+    assert_gpu_fallback_collect(
+            lambda spark : unary_op_df(spark, url_gen).selectExpr(
+                "a",
+                "parse_url(a, 'FILE')"
+                ),
+            'ParseUrl')
+
+@allow_non_gpu('ProjectExec', 'ParseUrl')
+def test_parse_url_authority_fallback():
+    assert_gpu_fallback_collect(
+            lambda spark : unary_op_df(spark, url_gen).selectExpr(
+                "a",
+                "parse_url(a, 'AUTHORITY')"
+                ),
+            'ParseUrl')
+
+@allow_non_gpu('ProjectExec', 'ParseUrl')    
+def test_parse_url_userinfo_fallback():
+    assert_gpu_fallback_collect(
+            lambda spark : unary_op_df(spark, url_gen).selectExpr(
+                "a",
+                "parse_url(a, 'USERINFO')"
+                ),
+            'ParseUrl')
+    
+def test_parse_url_too_many_args():
```

Comment:
This seems like a test of Spark itself rather than the plugin?  IIUC this error is going to be thrown not because the plugin throws it but during the query analysis which occurs before the plugin runs.

Reply:
Yes you are right! I removed this case and related unnecessary error handling code.